src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -1365,6 +1369,22 @@ private void Dispose(bool disposing)
         /// <param name="validationData">The validation data set.</param>
         public MulticlassPredictionTransformer<ImageClassificationModelParameters> Fit(
             IDataView trainData, IDataView validationData) => TrainTransformer(trainData, validationData);
+
+        [TlcModule.EntryPoint(Name = "Trainers.ImageClassifier",
+            Desc = "Train a ImageClassificationTrainer.",
+            UserName = UserName,
+            ShortName = ShortName)]
+        internal static CommonOutputs.MulticlassClassificationOutput TrainImageClassificationTrainer(IHostEnvironment env, Options input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register("TrainImageClassification");
+            host.CheckValue(input, nameof(input));
+            EntryPointUtils.CheckInputArgs(host, input);
+
+            return TrainerEntryPointsUtils.Train<Options, CommonOutputs.MulticlassClassificationOutput>(host, input,
+                () => new ImageClassificationTrainer(host, input),
+                () => TrainerEntryPointsUtils.FindColumn(host, input.TrainingData.Schema, input.LabelColumnName));
+        }
```

Comment:
Please follow the pattern in the codebase and put this entry point in its own class and consider giving a more appropriate name to entrypoint, TrainImageClassificationTrainer just sounds like a lot of redundancy

Reply:
This is following the pattern for the other trainers like NaiveBayesMulticlassTrainer, LbfgsMaximumEntropyMulticlassTrainer, LbfgsLogisticRegressionBinaryTrainer and so on, where the entrypoint is defined in the trainer class. The name TrainImageClassificationTrainer means "train an ImageClassificationTrainer", which is quite self-descriptive, but I can change it, although as much as I saw, it is not used to invoke the entrypoint anyway.

---
In reply to: [346132623](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346132623) [](ancestors = 346132623)

Reply:
Vast majority of entry points are defined in their own class, if some are part of the trainer that it is a deviation from the pattern. Please put this in this own class and make it consistent with the pattern. Have you seen any other entry point that starts with the pattern TrainXXXXTrainer? 

---
In reply to: [346459655](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346459655) [](ancestors = 346459655,346132623)

Reply:
From what I see, TrainMulticlassNaiveBayesTrainer, but that could be an anomaly too. 

---
In reply to: [346479753](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346479753) [](ancestors = 346479753,346459655,346132623)

Reply:
Yes, its an anomaly. I'm the author of NaiveBayes and it was a mistake to call it TrainMulticlassNaiveBayes. We must conform to patterns when possible, the person who reviewed my PR probably missed it. Implementing Naive Bayes was also the first change I made in this framework and I was only 1 week into the team when I put up the PR. That being said if you look at all the trainers you will see the pattern is "TrainBinary" or "TrainRanking" or "TrainKMeans" or "TrainSymSGD" ... this needs to be "TrainNaiveBayes" The fact that the word contains "Trainer" it already implies it will do the training so calling TrainerXXXTrainer seems incorrect English to me.

---
In reply to: [346484382](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346484382) [](ancestors = 346484382,346479753,346459655,346132623)

Reply:
Makes sense, changing.

---
In reply to: [346492366](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346492366) [](ancestors = 346492366,346484382,346479753,346459655,346132623)